internal/infrastructure/elastic_search/query_device_features.go

```diff
@@ -0,0 +1,141 @@
+package elasticsearch
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+type DeviceFeatures struct {
+	BatteryVoltage struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	FuelPercentRemaining struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Odometer struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Oil struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Soc struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Speed struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Tires struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+}
+
+type DeviceFeaturesResp struct {
+	Aggregations struct {
+		Features struct {
+			SumOtherDocCount int64 `json:"sum_other_doc_count"`
+			Buckets          []struct {
+				Key               string `json:"key"`
+				DocumentCount     int64  `json:"doc_count"`
+				DeviceDefinitions struct {
+					SumOtherDocCount int64 `json:"sum_other_doc_count"`
+					Buckets          []struct {
+						Key           string `json:"key"`
+						DocumentCount int64  `json:"doc_count"`
```

Comment:
```suggestion
						DocCount int64  `json:"doc_count"`
```
Just to match `SumOtherDocCount`. Not a big deal.